dbt_semantic_interfaces/validations/metrics.py

```diff
@@ -33,29 +34,62 @@ def _validate_cumulative_sum_metric_params(metric: Metric) -> List[ValidationIss
         issues: List[ValidationIssue] = []
 
         if metric.type == MetricType.CUMULATIVE:
-            if metric.type_params.window and metric.type_params.grain_to_date:
+            metric_context = MetricContext(
+                file_context=FileContext.from_metadata(metadata=metric.metadata),
+                metric=MetricModelReference(metric_name=metric.name),
+            )
+
+            for field in ("window", "grain_to_date"):
+                # Warn that the old type_params structure has been deprecated.
+                if getattr(metric.type_params, field):
+                    issues.append(
+                        ValidationWarning(
+                            context=metric_context,
+                            message=(
+                                f"Cumulative `type_params.{field}` field has been moved and will soon be deprecated. "
+                                f"Please nest that value under `type_params.cumulative_type_params.{field}`."
+                            ),
+                        )
+                    )
+                # Warn that window or grain_to_date is duplicated.
+                if (
+                    getattr(metric.type_params, field)
+                    and metric.type_params.cumulative_type_params
+                    and getattr(metric.type_params.cumulative_type_params, field)
+                ):
+                    issues.append(
+                        ValidationError(
+                            context=metric_context,
+                            message=(
+                                f"`{field}` set twice in cumulative metric '{metric.name}'. "
+                                "Please remove it from `type_params`."
+                            ),
+                        )
+                    )
```

Comment:
We actually want them to be duplicated I think. In which case we should change this to error if the fields are different.